### PR TITLE
chore(schema): sql performance improvement for explorative analysitcs

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

export enum ExplorePathNodeType {
  EVENT = 'event',
  PAGE_TITLE = '_page_title',
  PAGE_URL = '_page_url',
  SCREEN_NAME = '_screen_name',
  SCREEN_ID = '_screen_id',
}
--->
export enum ExplorePathNodeType {
  EVENT = 'event',
  PAGE_TITLE = 'page_title',
  PAGE_URL = 'page_url',
  SCREEN_NAME = 'screen_name',
  SCREEN_ID = 'screen_id',
}

export enum ConditionCategory {
  USER_OUTER = 'user_outer',
  USER = 'user',
  EVENT = 'event',
  DEVICE = 'device',
  GEO = 'geo',
  APP_INFO = 'app_info',
  TRAFFIC_SOURCE = 'traffic_source',
  OTHER = 'other',
}

--->

export enum ConditionCategory {
  USER_OUTER = 'user_outer',
  USER = 'user',
  EVENT = 'event',
  EVENT_OUTER = 'event_outer',
}

EVENT_OUTER = original's OTHER

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/common/explore-types.ts [hunk 1]
```diff
@@ -87,10 +87,10 @@ export enum ExplorePathSessionDef {
 
 export enum ExplorePathNodeType {
   EVENT = 'event',
-  PAGE_TITLE = '_page_title',
-  PAGE_URL = '_page_url',
-  SCREEN_NAME = '_screen_name',
-  SCREEN_ID = '_screen_id',
+  PAGE_TITLE = 'page_title',
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/common/explore-types.ts:90]: I see the duplicated `ExplorePathNodeType` in `frontend/src/ts/explore-types.ts`.

We need extract them to base lib.

**Review by zxkane — CHANGES_REQUESTED** [review]

**Review by tyyzqmf — APPROVED** [review]

**Review by dengmingtong — APPROVED** [review]

File: src/control-plane/backend/lambda/api/test/api/metadata.test.ts [hunk 2]
```diff
@@ -1188,7 +1188,7 @@ describe('Metadata Event test V2', () => {
       projectId: MOCK_PROJECT_ID,
       appId: MOCK_APP_ID,
       name: MOCK_EVENT_PARAMETER_NAME,
-      category: ConditionCategory.APP_INFO,
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/test/api/metadata.test.ts:1191]: Why not use the constant?

File: src/control-plane/backend/lambda/api/test/api/metadata.test.ts [hunk 3]
```diff
@@ -1493,10 +1493,10 @@ describe('Metadata Event Attribute test', () => {
       Items: [],
     });
     const res = await request(app)
-      .get(`/api/metadata/event_parameter?projectId=${MOCK_PROJECT_ID}&appId=${MOCK_APP_ID}&name=${MOCK_EVENT_PARAMETER_NAME}&category=${ConditionCategory.APP_INFO}&type=${MetadataValueType.STRING}`);
+      .get(`/api/metadata/event_parameter?projectId=${MOCK_PROJECT_ID}&appId=${MOCK_APP_ID}&name=${MOCK_EVENT_PARAMETER_NAME}&category=app_info&type=${MetadataValueType.STRING}`);
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/test/api/metadata.test.ts:1496]: Why not use the constant?

File: src/control-plane/backend/lambda/api/test/api/metadata.test.ts (hunk 2 above)
Comment (llmin) [reply src/control-plane/backend/lambda/api/test/api/metadata.test.ts:1191]: fixed

File: src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts [hunk 4]
```diff
@@ -593,6 +557,7 @@ export function buildEventPropertyAnalysisView(sqlParameters: SQLParameters) : s
   });
 }
 
+//todo event_date
```
[the remark this answers is not in the document]
Comment (llmin) [reply src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts]: remove

**Review by zxkane — APPROVED** [review]

**Pull request merged**